### DISCUSSION TITLE
How to define wix variables as part of msbuild task in wix4

### DISCUSSION THREAD

**Discussion by svg2003** [wixtoolset/issues#7795] [Questions] [answered]
Hi,

In wix3 there is ability to add new wix variables inside msbuild task as following:

  <Target Name="SomeCustomTask">

    <CreateProperty Value="MyNewWixVariable="some value";$(WixVariables)">
      <Output TaskParameter="Value" PropertyName="WixVariables" />
    </CreateProperty>
  </Target>

And after use its value in wxs file like below:

<Payload SourceFile="!(wix.MyNewWixVariable)" Name="some file name" />

In wix4 such ability seems missing. Can you please advice alternatives how to do it, or restore previous functionality back?

**Comment by robmen** [comment, marked as answer]
Define your `WixVariables` property in your .wixproj or use the `BindVariable` item instead.
